Makefile

```diff
@@ -49,3 +49,28 @@ coverage:
 coveralls: coverage
 	go get github.com/mattn/goveralls
 	goveralls -coverprofile=$(COVERAGE_FILE) -service=github
+
+DURATION ?= 3s
+COUNT ?= 5
+
+.PHONY: bench
+bench:
+	rm -rf bench
+	mkdir bench
+	go clean -testcache
+	go test -bench=. -benchmem -benchtime=${DURATION} -count=${COUNT} 2>&1 | tee bench/bench.txt
+	cd bench && git clone https://go.googlesource.com/perf && cd perf && go install ./cmd/benchstat
+	benchstat bench/bench.txt
+	rm -rf bench/perf
+
+.PHONY: perf-diff
+perf-diff:
```

Comment:
I would split this target for two:
- `perf-reference` - run tests on master branch and put results to a file `reference.txt`. We need to do it only once and reuse it for further testing
- `perf-diff` - depends on file `reference.txt` produced by `perf-reference` and target `bench`

Reply:
Good idea, fixed.